git-server/src/hooks/post_receive.rs

```diff
@@ -349,12 +349,23 @@ impl PostReceive {
         };
         println!("Running custom receive hook...");
 
-        let mut child = Command::new(hook)
+        let child = Command::new(hook)
             .stderr(Stdio::inherit())
             .stdout(Stdio::inherit())
             .stdin(Stdio::piped())
             .spawn()
-            .map_err(Error::CustomHook)?;
+            .map_err(|err| match err.kind() {
+                ErrorKind::NotFound => Error::Io(err),
+                _ => Error::CustomHook(err),
+            });
```

Comment:
There's no need to change this, `CustomHook` has the `io::Error` inside of it, so you can just match on it and on the error-kind in the conditional below.

Reply:
https://github.com/radicle-dev/radicle-client-services/pull/113/commits/dfcf3f0f42da80638c52a349ece6e8a233f6bcaa#diff-bcec911f4891588d789f1bf73e0dadb81401adba787874fd6a3eced15bc552d7R359-R364